core/src/main/java/bio/terra/pearl/core/service/survey/SurveyService.java

```diff
@@ -139,6 +139,13 @@ public Survey createNewVersion(UUID portalId, Survey survey) {
         return create(newSurvey);
     }
 
+    public void attachAnswerMappings(Survey survey) {
+        survey.getAnswerMappings().clear();
+        survey.getAnswerMappings().addAll(
+                answerMappingDao.findBySurveyId(survey.getId())
```

Comment:
Why `clear` and `addAll` instead of `setAnswerMappings`? I don't think there's anything special about the collection instances (like there could be in a Hibernate/JPA application).

Reply:
yup, that is definitely my hibernate muscle-memory showing